### PR TITLE
Update blog to use `mem` feature of `compiler_builtins`

### PR DESCRIPTION
**Pull request by phil-opp**
Updates the blog for the code changes introduced in #865.﻿

### CODE REVIEW

File: blog/content/second-edition/posts/02-minimal-rust-kernel/index.md [hunk 1]
```diff
@@ -289,32 +289,31 @@ After setting the `unstable.build-std` configuration key and installing the `rus
...
 
-While we didn't need any of these functions to compile our kernel right now, they will be required as soon as we add some more code to it. So it's a good idea to provide implementations for these functions now to avoid linker errors later. While there is no way to enable the implementations of the `compiler_builtins` crate yet (see the [tracking issue](https://github.com/rust-lang/wg-cargo-std-aware/issues/15)), there is a good alternative: the [`rlibc`] crate.
+Since we can't link to the C library of the operating system, we need an alternative way to provide these functions to the compiler. One possible approach for this could be to implement our own `memset` etc. functions and apply the `#[no_mangle]` attribute to them (to avoid the atomatic renaming during compilation). However, this is dangerous since the slightest mistake in the implementation of these functions could lead to undefined behavior. So it's a good idea to reuse existing well-tested implementations instead.
```
Comment (robert-w-gries) [inline blog/content/second-edition/posts/02-minimal-rust-kernel/index.md:296]: Typo: `s/atomatic/automatic`

Comment (phil-opp) [reply blog/content/second-edition/posts/02-minimal-rust-kernel/index.md]: Thanks, fixed!

**Comment by briankung** [comment]
Hi, I know you're in the middle of a big rewrite, but I'm getting a linking error using this feature instead of `rlibc` (click to expand the compiler error) with the current code. I just made it to [Try it out!](https://os.phil-opp.com/vga-text-mode/#try-it-out) where we try to `vga_buffer::print_something()` for the first time.

<details>
  <summary><code>rust-lld: error: undefined symbol: memset</code> (click for more...)</summary>
  
  ```
       Compiling blog_os v0.1.0 (/Users/brian/Code/local/rust/blog_os)
    error: linking with `rust-lld` failed: exit code: 1
      |
      = note: "rust-lld" "-flavor" "gnu" "--eh-frame-hdr" "-L" "/Users/brian/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-blog_os/lib" "/Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/blog_os-655c79911b6614f8.1acu7aet213x3h7.rcgu.o" "/Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/blog_os-655c79911b6614f8.25wnobk433yf0zoc.rcgu.o" "/Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/blog_os-655c79911b6614f8.2ghhqknm3xz7jve5.rcgu.o" "/Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/blog_os-655c79911b6614f8.2op71tmzrz9p7xm4.rcgu.o" "/Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/blog_os-655c79911b6614f8.310auomji65262sd.rcgu.o" "/Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/blog_os-655c79911b6614f8.4y1p3058sfp137pt.rcgu.o" "/Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/blog_os-655c79911b6614f8.502yzro07n60f3y8.rcgu.o" "/Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/blog_os-655c79911b6614f8.pfjvfcwkxqoh6m8.rcgu.o" "-o" "/Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/blog_os-655c79911b6614f8" "--gc-sections" "-L" "/Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps" "-L" "/Users/brian/Code/local/rust/blog_os/target/debug/deps" "-L" "/Users/brian/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/x86_64-blog_os/lib" "-Bstatic" "/Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/librustc_std_workspace_core-c85c4f48002fcc96.rlib" "/Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/libcore-d6cd1565a3612d4c.rlib" "/Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/libcompiler_builtins-e7f2415038ad1aa4.rlib" "-Bdynamic"
      = note: rust-lld: error: undefined symbol: memset
              >>> referenced by num.rs:213 (/Users/brian/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/fmt/num.rs:213)
              >>>               core-d6cd1565a3612d4c.core.5np4my57-cgu.13.rcgu.o:(core::fmt::num::imp::fmt_u64::hb588b66b54ea6795) in archive /Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/libcore-d6cd1565a3612d4c.rlib

              rust-lld: error: undefined symbol: memcpy
              >>> referenced by intrinsics.rs:1860 (/Users/brian/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/intrinsics.rs:1860)
              >>>               core-d6cd1565a3612d4c.core.5np4my57-cgu.14.rcgu.o:(core::intrinsics::copy_nonoverlapping::h5da078f0cb1de4c4) in archive /Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/libcore-d6cd1565a3612d4c.rlib
              >>> referenced by intrinsics.rs:1860 (/Users/brian/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/intrinsics.rs:1860)
              >>>               core-d6cd1565a3612d4c.core.5np4my57-cgu.14.rcgu.o:(core::intrinsics::copy_nonoverlapping::h9204dda9829016e4) in archive /Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/libcore-d6cd1565a3612d4c.rlib
              >>> referenced by intrinsics.rs:1860 (/Users/brian/.rustup/toolchains/nightly-x86_64-apple-darwin/lib/rustlib/src/rust/library/core/src/intrinsics.rs:1860)
              >>>               core-d6cd1565a3612d4c.core.5np4my57-cgu.14.rcgu.o:(core::intrinsics::copy_nonoverlapping::hb37a74d76009075d) in archive /Users/brian/Code/local/rust/blog_os/target/x86_64-blog_os/debug/deps/libcore-d6cd1565a3612d4c.rlib
              >>> referenced 1 more times


    error: aborting due to previous error

    error: could not compile `blog_os`

    To learn more, run the command again with --verbose.
  ```
</details>

I made sure to update my rust nightly toolchain this time:

```
$ rustc --version
rustc 1.49.0-nightly (a1dfd2490 2020-10-05)
```

**Comment by robert-w-gries** [comment]
I'm able to build `post-03` branch using that same nightly compiler.

Are you sure you're building with the `compiler-builtins-mem` feature? I would check the `.cargo/config.toml` file and make sure however you're building the kernel is pulling in that enabled feature.

In case you were wondering, I did the following to test it:

```
cd blog_os
git checkout post-03
cargo build
```

**Comment by robert-w-gries** [comment]
This might be a little more helpful, try enabling the feature in your `cargo run` command like so:

```
cargo run -Zbuild-std-features=compiler-builtins-mem
```

**Comment by briankung** [comment]
That did the trick! And I see what I got wrong - I was reading the diff and put the config into `Cargo.toml`. Thanks!

**Pull request merged**